### PR TITLE
Add `augerctl get`

### PR DESCRIPTION
**Pull request by wzshiming**
part of #3
part of #62 

### Preparation cluster
Create a cluster and expose etcd port, to facilitate the creation of a cluster using kwokctl, this can be any other k8s cluster
``` bash
# brew install kwok
kwokctl create cluster --etcd-port 2379
```

### Get a single resource

Get the kubernetes.default service

``` bash
go run ./augerctl get services -n default kubernetes
```

### Get the all of the resource

``` bash
go run ./augerctl get leases -n kube-system
```

### Get the all of the etcd
``` bash
go run ./augerctl get
```

### CODE REVIEW

File: pkg/augerctl/ctl.go
Comment (jmhbnz) [inline pkg/augerctl/ctl.go:1]: I would lean towards aligning with structure of etcd `etcdctl` and `etcdutl` where commands are subdirectory of `cmd` rather than `pkg`, i.e:
- https://github.com/etcd-io/etcd/tree/main/etcdctl/ctlv3/command
- https://github.com/etcd-io/etcd/tree/main/etcdutl/etcdutl

cc @siyuanfoundation for second opinion before changing it.

File: pkg/augerctl/get_command.go
Comment (jmhbnz) [inline pkg/augerctl/get_command.go:42]: Suggest adding some [examples](https://github.com/spf13/cobra/blob/v1.8.1/command.go#L80). Not a blocker though we can do it as a follow-up if you prefer.

File: cmd/augerctl/main.go
Comment (jmhbnz) [inline cmd/augerctl/main.go:1]: Suggest adding a Markdown file at this level outlining the intent for `augerctl` and basic usage. Can re-use example for this pr description.

**Review by jmhbnz — COMMENTED** [review]
Thanks for raising this @wzshiming - Below are some initial suggestions, nothing major.

**Comment by wzshiming** [comment]
Done

File: cmd/augerctl/README.md
Comment (siyuanfoundation) [inline cmd/augerctl/README.md:10]: instructions of how to build it.

File: cmd/augerctl/README.md
Comment (siyuanfoundation) [inline cmd/augerctl/README.md:17]:
```suggestion
+ a comma-delimited list of machine addresses in the etcd cluster
```

File: cmd/augerctl/README.md
Comment (siyuanfoundation) [inline cmd/augerctl/README.md:52]:
```suggestion
List a single service with namespace `default` and name `kubernetes`
```

File: cmd/augerctl/README.md
Comment (siyuanfoundation) [inline cmd/augerctl/README.md:58]:
```suggestion
List a single resource of type `priorityclasses` and name `system-node-critical`  without namespaced
```

File: cmd/augerctl/README.md
Comment (siyuanfoundation) [inline cmd/augerctl/README.md:50]: can you add some explanation of how this is different from `kubectl`?

File: cmd/augerctl/main.go
Comment (siyuanfoundation) [inline cmd/augerctl/main.go:1]: structure-wise, this loooks very weird, because `augerctl` is not a subcommand of auger, but it is inside `cmd/`. There are two ways we can make it cleaner:
1. instead of `augerctl`, just add it as a subcommand `auger ctl`. Or
2. refactor the main functions to a separate package, and have `augerctl` and `auger` modules at the top level. 

@wzshiming @jmhbnz WDYT?

Comment (jmhbnz) [reply cmd/augerctl/main.go:1]: I would lean towards having a separate package similar to our `etcd` / `etcdctl` model.  With that said it's easier to implement 1 and I'm not necessarily opposed to that option if we want to start with that.

Comment (siyuanfoundation) [reply cmd/augerctl/main.go:1]: I also think a model similar to our `etcd` / `etcdctl` model is more extendable in the long term.

File: augerctl/command/get_command.go
Comment (siyuanfoundation) [inline augerctl/command/get_command.go:29]: should there be an option to limit the number of results?

File: augerctl/command/get_command.go
Comment (siyuanfoundation) [inline augerctl/command/get_command.go:37]: Add an example with group and resource as well.

File: augerctl/command/ctl.go
Comment (wzshiming) [reply augerctl/command/ctl.go:50]: https://github.com/etcd-io/etcd/blob/e34a10191473e5be14ca3b6f972327f28e9b20db/etcdctl/ctlv3/ctl.go#L71

I expect this arg to be the same as etcdctl

File: pkg/client/client_get.go
Comment (wzshiming) [reply pkg/client/client_get.go:58]: https://github.com/etcd-io/auger/blob/6d4429e755661e17dcbdb998c92bcf691230c7d1/pkg/client/client_get.go#L86-L89


This is to avoid adding the Rev option again

File: augerctl/command/get_command.go
Comment (wzshiming) [reply augerctl/command/get_command.go:29]: Good point, but `kubectl` doesn't have this yet, maybe we should discuss this after the PR merge.

File: augerctl/command/ctl.go
Comment (siyuanfoundation) [reply augerctl/command/ctl.go:49]: what about this one?

**Comment by wzshiming** [comment]
done

File: pkg/client/util.go
Comment (siyuanfoundation) [inline pkg/client/util.go:75]: What does wantSingle mean? Does it mean if name is specified, we expect a single result?

**Comment by siyuanfoundation** [comment]
1 minor comment. Otherwise lgtm

File: pkg/client/util.go
Comment (wzshiming) [reply pkg/client/util.go:75]: Ah, you're right, it was changed by mistake by the global replacement

**Review by siyuanfoundation — APPROVED** [review]

**Review by jmhbnz — APPROVED** [review]
LGTM - Many thanks for your patience with review, and for all your work on this @wzshiming

**Pull request merged**
